app/models/concerns/balance.rb

```diff
@@ -22,22 +37,19 @@ def new_outstanding_balance
     end
   end
 
-  # This method is the one we're gradually replacing with `#new_outstanding_balance`. Having them
-  # separate enables us to choose which implementation we want depending on the context using
-  # a feature toggle. This avoids incosistent behavior across the app during that incremental
-  # refactoring.
-  #
-  # It is meant to be removed as soon as we get product approval that the new implementation has
-  # been working correctly in production.
-  def outstanding_balance
-    total - payment_total
-  end
-
   def outstanding_balance?
+    print_deprecation(__method__)
     !outstanding_balance.zero?
   end
 
   def display_outstanding_balance
-    Spree::Money.new(outstanding_balance, currency: currency)
+    print_deprecation(__method__)
+    outstanding_balance.amount
+  end
+
+  def print_deprecation(method)
+    ActiveSupport::Deprecation
+      .warn("#{method} is soon to be removed in favor of OrderBalance. \
+            See https://github.com/openfoodfoundation/openfoodnetwork/pull/7064.")
```

Comment:
I could imagine that these warnings are quite annoying, filling up production logs. But okay. It would be helpful to describe how to avoid this warning, for example adding `BETA_TESTERS: "all"` to your application.yml.

Won't our tests trigger this deprecation all the time? In my opinion, deprecations are good for things you want others not to use but first you have to stop using it yourself.

Reply:
Fair enough. I can find the way to silence it by default and enbale it manually when in development and so it will still be useful to remove calls to it.

Reply:
I decided it's not worth adding these warnings. I can copy&paste this code when I start cleaning dead code to find the callers without having to push it to the repo. I think it's the simplest approach.

Reply:
I found https://www.fastruby.io/blog/rails/upgrades/deprecation-warnings-rails-guide.html a quite interesting read on the topic.

Reply:
In another project we are only on Rails 4.2 now but we set deprecation warnings to raise in specs and by getting rid of all of them I feel very close to Rails 5.0 already.